service/lib/agama/software/manager.rb

```diff
@@ -140,12 +142,16 @@ def probe
         progress.step(_("Refreshing repositories metadata")) { repositories.load }
         progress.step(_("Calculating the software proposal")) { propose }
 
-        Yast::Stage.Set("initial")
         update_issues
       end
 
-      def initialize_target_repos
-        Yast::Pkg.TargetInitialize("/")
+      def initialize_target
+        # create the zypp lock also in the target directory
+        ENV["ZYPP_LOCKFILE_ROOT"] = TARGET_DIR
+        # cleanup the previous content (after service restart or crash)
+        FileUtils.rm_rf(TARGET_DIR)
+        FileUtils.mkdir_p(TARGET_DIR)
+        Yast::Pkg.TargetInitialize(TARGET_DIR)
```

Comment:
I was wondering whether this could break the actual RPM installation, but apparently it's OK as `TargetInitialize` will be also called with the destination directory later:
```console
60eaee65c80e:/ # grep TargetInit /var/log/YaST2/y2log
2024-06-13 12:31:25 <1> 60eaee65c80e(2007) [Pkg] software/manager.rb:154 Pkg Builtin called: TargetInitialize
2024-06-13 12:33:59 <1> 60eaee65c80e(2007) [Pkg] software/proposal.rb:157 Pkg Builtin called: TargetInitialize
2024-06-13 12:35:06 <1> 60eaee65c80e(2007) [Pkg] software/proposal.rb:157 Pkg Builtin called: TargetInitialize
2024-06-13 12:36:07 <1> 60eaee65c80e(2007) [Pkg] software/proposal.rb:157 Pkg Builtin called: TargetInitialize
```

Reply:
Yes, later the target is later moved to `/mnt`. We can do this *only after* the installation is started, after the partitioning step.